### PR TITLE
feat: add dockerfile to kakarot-rpc

### PR DESCRIPTION
**Pull request by Eikix**
Adding Docker to our project in order to facilitate Paas deployment

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# Build the application
+RUN cargo build --all --release
+
+# Create a new container from scratch to reduce image size
+FROM debian:buster-slim
+
+# Install any necessary dependencies
+RUN apt-get update && apt-get install -y libssl-dev ca-certificates && rm -rf /var/lib/apt/lists/*
+
+# RUN update-ca-certificates
```
Comment (ClementWalter) [inline Dockerfile:18]: remove?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+WORKDIR /usr/src/app
+
+# Copy the built binary from the previous container
+COPY --from=builder /usr/src/rpc/target/release/kakarot-rpc .
+
+# Expose the port that the RPC server will run on
+EXPOSE 3030
+
+# Run the binary
+CMD ["./kakarot-rpc"]
```
Comment (ClementWalter) [inline Dockerfile:30]: though this is fine, there is apparently a best practice consisting in using `tiny`, see for example https://github.com/eqlabs/pathfinder/blob/4ab915a830953ed6f02af907937b46cb447d9a92/Dockerfile#L120

File: rpc/src/lib.rs [hunk 3]
```diff
@@ -20,7 +20,7 @@ pub async fn run_server(
     starknet_client: Box<dyn KakarotClient>,
 ) -> Result<(SocketAddr, ServerHandle), RpcError> {
     let server = ServerBuilder::default()
-        .build("127.0.0.1:3030".parse::<SocketAddr>()?)
+        .build("0.0.0.0:3030".parse::<SocketAddr>()?)
```
Comment (ClementWalter) [inline rpc/src/lib.rs:23]: should we use a ENV var for the port ?

File: Dockerfile (hunk 2 above)
Comment (Eikix) [reply Dockerfile:30]: ok!

File: rpc/src/lib.rs (hunk 3 above)
Comment (Eikix) [reply rpc/src/lib.rs:23]: ok!

**Review by ClementWalter — APPROVED** [review]

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**